src/main/java/org/truffleruby/core/regexp/RegexpOptions.java

```diff
@@ -153,7 +153,7 @@ public Encoding setup(RubyContext runtime) {
             return EUCJPEncoding.INSTANCE;
         } else if (explicitKCode == KCode.SJIS) {
             setFixed(true);
-            return Layouts.ENCODING.getEncoding(runtime.getEncodingManager().getRubyEncoding(WINDOWS31J));
+            return EncodingManager.getEncoding(WINDOWS31J);
```

Comment:
I think we should remove the RubyContext argument of `setup()` if it's no longer needed.